### ISSUE TITLE
Cursor jumps on multiline text inputs

### ISSUE THREAD

**Issue by AdrianFahrbach** [microsoft/react-native-macos#2127]
### Environment

```markdown
react-native -v: v0.73.8
npm ls react-native-macos: v0.73.26
node -v: v20.12.1
npm -v: v10.5.0
yarn --version: not in use
xcodebuild -version: Xcode 15.4 Build version 15F31d
```

### Steps to reproduce the bug

1. Create a TextInput component
2. Set its multiline prop to true
3. Type in a few words and blur the input
4. Now click on it again, somewhere in the middle of the text
5. Type something and your cursor should jump to the end

### Expected Behavior

The cursor should keep its position.

### Actual Behavior

In the attached video I'm testing it with multiline true and false, two times each.

https://github.com/microsoft/react-native-macos/assets/45072099/955cc36a-8b48-4523-90cf-827873bd0c13

### Reproducible Demo

You could test this in our [Jira Time Tracker](https://github.com/AdrianFahrbach/jira-time-tracker) repository.
However to make this work a `.env` file with a Jira API key as well as a Jira account are required. Send me a mail to adrianfahrbach@me.com and I can provide you with both.

### Additional context

This _could_ be related to #2066 and/or #2090.

**Comment by Saadnajmi** [comment]
Tried running Jira Time tracker through test flight, but don't have a Jira account (We use Azure Devops here at Microsoft :P), so I can't easily repro (will email though).

For the TextInput, are you setting the `value` property on it?

**Comment by AdrianFahrbach** [comment]
> For the TextInput, are you setting the value property on it?

Yes, I'm setting the value through a state. The input in question is [this one](https://github.com/AdrianFahrbach/jira-time-tracker/blob/main/src/screens/EditWorklog.tsx#L96).
Just answered you mail btw!

**Comment by Saadnajmi** [comment]
I _think_ this was fixed in React Native upstream on the latest versions. There was a big PR around whose link I can dig up if need be. Closing unless still an issue.
